src/audio/base_fw.c

```diff
@@ -228,13 +235,99 @@ static int basefw_mem_state_info(uint32_t *data_offset, char *data)
 	return 0;
 }
 
+static uint32_t basefw_set_system_time(uint32_t param_id,
+				       bool first_block,
+				       bool last_block,
+				       uint32_t data_offset,
+				       char *data)
+{
+	if (!(first_block && last_block))
+		return IPC4_INVALID_REQUEST;
+
+	global_system_time_info.host_time.val_l = (*(struct ipc4_system_time *)(data)).val_l;
+	global_system_time_info.host_time.val_u = (*(struct ipc4_system_time *)(data)).val_u;
+	uint64_t current_dsp_time = sof_cycle_get_64();
```

Comment:
@piotrkmx @mwasko @abonislawski Can you help me out here. I don't understand why we set global_system_time_info (and use SRAM to store this in DSP global memory) if it's only used by host to ask for host time. This makes no sense as host already knows its own time. 

Probably the intent is that we should configure Zephyr's system time to global_system_time_info (or only Zephyr logging subsystem's time), so this action would also has some impact to DSP operation. Is there other purpose than affecting logging timestamps? That will impact whether we should set Zephyr's kernel time, or is it enough to set timing offset for the logging subsystem. Based on my initial review, setting the kernel time might be more complex (I see no such interfaces, need to consult Zephyr folks. adding @nashif to the loop here ).

Reply:
Or is this only set to calculate the ART/RTC value later in ext_get_system_time()?

Reply:
@kv2019i the usage is very limited in this PR but it can be used in another PR for logging timestamp offset, e.g using log_set_timestamp_func() which already exists in zephyr.
I think we dont want to change zephyr kernel time.

Reply:
@kv2019i 
Yes, before ext get system time system time can be used, it must be set via system time. 

Of course the host knows it's host time. If I understand correctly, the purpose of this is to let Zephyr know the host time. As Zephyr system time can be different from host system time. So it can use this information to produce logs that include not Zephyr's kernel time, but with host system time.  The other usage mentioned besides logs, is probes packets. 

I personally do not see the point to change the whole Zephyr's kernel time, so firmware can produce logs or any other event timestamps that contain in themselves the current host system time. As we discussed it today within our team, it may have way to much implications.

Reply:
Ack thanks. I do agree that changing the kernel looks like potentially high-effort and strictly not needed. @piotrkmx can you add a "/* TODO: configurate time to logging subsystem */" which would make this clearer in the code.